psc-bundle/Main.hs

```diff
@@ -30,32 +35,42 @@ import qualified Options.Applicative as Opts
 
 import qualified Paths_purescript as Paths
 
+import SourceMap
+import SourceMap.Types
+
 -- | Command line options.
 data Options = Options
   { optionsInputFiles  :: [FilePath]
   , optionsOutputFile  :: Maybe FilePath
   , optionsEntryPoints :: [String]
   , optionsMainModule  :: Maybe String
   , optionsNamespace   :: String
+  , optionsSourceMaps  :: Bool
   } deriving Show
 
 -- | The main application function.
 -- This function parses the input files, performs dead code elimination, filters empty modules
 -- and generates and prints the final Javascript bundle.
-app :: (MonadError ErrorMessage m, MonadIO m) => Options -> m String
+app :: (MonadError ErrorMessage m, MonadIO m) => Options -> m (Maybe SourceMapping, String)
 app Options{..} = do
   inputFiles <- concat <$> mapM (liftIO . glob) optionsInputFiles
   when (null inputFiles) . liftIO $ do
     hPutStrLn stderr "psc-bundle: No input files."
     exitFailure
+  when (isNothing optionsOutputFile && optionsSourceMaps == True) . liftIO $ do
+    hPutStrLn stderr "psc-bundle: Source maps only supported when output file specified."
+    exitFailure
+
   input <- for inputFiles $ \filename -> do
     js <- liftIO (readUTF8File filename)
     mid <- guessModuleIdentifier filename
-    length js `seq` return (mid, js)                                            -- evaluate readFile till EOF before returning, not to exhaust file handles
+    length js `seq` return (mid, filename, js)                                            -- evaluate readFile till EOF before returning, not to exhaust file handles
 
   let entryIds = map (`ModuleIdentifier` Regular) optionsEntryPoints
 
-  bundle input entryIds optionsMainModule optionsNamespace
+  currentDir <- liftIO getCurrentDirectory
+  let outFile = if optionsSourceMaps then fmap (currentDir </>) optionsOutputFile else Nothing
+  bundleSM input entryIds optionsMainModule optionsNamespace outFile
```

Comment:
Guess I don't know what I built after making changes but it wasn't psc-bundle! Fixed  & squashed.